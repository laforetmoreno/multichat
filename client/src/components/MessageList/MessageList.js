import React from 'react';
import { node } from 'prop-types';

import Message from '../Message';

import styles from './MessageList.module.scss';

const MessageList = ({ messages, user }) => (
  <ul data-testid="list" className={styles.wrapper}>
    {messages.map((message, index) => (
      <Message user={user} key={index} message={message} />
    ))}
  </ul>
);

MessageList.propTypes = {
  children: node,
};

export default MessageList;
