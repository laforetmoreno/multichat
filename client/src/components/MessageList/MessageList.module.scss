.wrapper {
  margin: 20px 0;
  padding: 0;
  text-indent: 0;
  list-style: none;
  width: 100%;
}
