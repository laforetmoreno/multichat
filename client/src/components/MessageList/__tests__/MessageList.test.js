import React from 'react';
import MessageList from '../MessageList';
import renderer from 'react-test-renderer';

describe('<MessageList />', () => {

  const messagesFake = [
    { value: 'hi', author: 'joe' }
  ]

  const userFake = {
    nickName: 'joe joe'
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<MessageList messages={messagesFake} user={userFake} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});