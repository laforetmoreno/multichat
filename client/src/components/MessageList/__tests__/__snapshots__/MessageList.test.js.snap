// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageList /> renders correctly 1`] = `
<ul
  className="wrapper"
  data-testid="list"
>
  <li
    className="item"
  >
    <div
      className="messageContainer false"
    >
      <strong
        className="author"
      >
        joe
      </strong>
      <span
        className="value"
      >
        hi
      </span>
    </div>
  </li>
</ul>
`;
